test/Kubernetes/KubernetesApiExtensionsTests.cs

```diff
@@ -2,11 +2,65 @@
 using k8s.Models;
 using Snd.Sdk.Kubernetes;
 using Xunit;
+using System;
+using System.Linq;
 
 namespace Snd.Sdk.Tests.Kubernetes;
 
 public class KubernetesApiExtensionsTests
 {
+    [Theory]
+    [InlineData("Restart", "Restart", new int[] { 1, 2 }, new int[] { 2, 3 }, 3)]
+    [InlineData("Restart", "Fail", null, null, null)] // Expecting an exception
+    public void MergePodFailurePolicyRules_Test(string actionA, string actionB, int[] exitCodesA, int[] exitCodesB, int? expectedExitCodesCount)
+    {
+        var ruleA = new V1PodFailurePolicyRule(actionA,
+            new V1PodFailurePolicyOnExitCodesRequirement("In", exitCodesA?.ToList()));
+
+        var ruleB = new V1PodFailurePolicyRule(actionB,
+            new V1PodFailurePolicyOnExitCodesRequirement("In", exitCodesB?.ToList()));
+
+        if (expectedExitCodesCount.HasValue)
+        {
+            var result = KubernetesApiExtensions.MergePodFailurePolicyRules(ruleA, ruleB);
+
+            Assert.Equal(actionA, result.Action);
```

Comment:
multi-assert tests are bad practice. You should test one thing - a final object in this case, that you should construct from expected values

Reply:
That is good to know, I was using multiple assertions thinking that I was doing a great thing. I'll have that in mind going forward. Updated it now in:  [d21279a](https://github.com/SneaksAndData/esd-services-sdk/pull/75/commits/d21279af3c03081c2504d45213df2471774025cd)